src/main/java/frc/robot/RobotContainer.java

```diff
@@ -381,4 +385,18 @@ private void configureBindings() {
   public Command getAutonomousCommand() {
     return autoChooser.getSelected();
   }
+
+  /**
+   * Gets the current alliance. Returns null if there is no alliance.
+   *
+   * @return the current alliance. Nullable.
+   */
+  public static @Nullable Alliance getAlliance() {
```

Comment:
This is boilerplate that defeats the purpose of creating it in the first place. We want a boolean that is true if we know we're on red alliance, and returns false in all other circumstances (including if we don't know).